specification/alertsmanagement/resource-manager/Microsoft.AlertsManagement/preview/2019-03-01-preview/AlertsManagement.json

```diff
@@ -0,0 +1,2157 @@
+          },
+          {
+            "$ref": "#/parameters/monitorService"
+          },
+          {
+            "$ref": "#/parameters/monitorCondition"
+          },
+          {
+            "$ref": "#/parameters/severity"
+          },
+          {
+            "$ref": "#/parameters/smartGroupState"
+          },
+          {
+            "$ref": "#/parameters/timeRange"
+          },
+          {
+            "$ref": "#/parameters/pageCount"
+          },
+          {
+            "$ref": "#/parameters/smartGroupsSortBy"
+          },
+          {
+            "$ref": "#/parameters/sortOrder"
+          },
+          {
+            "$ref": "#/parameters/api-version"
+          }
+        ],
+        "produces": [
+          "application/json"
+        ],
+        "responses": {
+          "200": {
+            "description": "OK. Returns list of all smartGroups.",
+            "schema": {
+              "$ref": "#/definitions/smartGroupsList",
+              "description": "List of smart groups in value property."
+            }
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "#/definitions/errorResponse"
+            }
+          }
+        },
+        "x-ms-examples": {
+          "List": {
+            "$ref": "./examples/SmartGroups_List.json"
+          }
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/providers/Microsoft.AlertsManagement/smartGroups/{smartGroupId}": {
+      "get": {
+        "operationId": "SmartGroups_GetById",
+        "summary": "Get information of smart alerts group.",
+        "description": "Get details of smart group.",
+        "produces": [
+          "application/json"
+        ],
+        "parameters": [
+          {
+            "$ref": "#/parameters/subscriptionId"
+          },
+          {
+            "$ref": "#/parameters/smartGroupId"
+          },
+          {
+            "$ref": "#/parameters/api-version"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK. Returns the group with the specified smart group Id.",
+            "headers": {
+              "x-ms-request-id": {
+                "type": "string",
+                "description": "Service generated Request ID."
+              }
+            },
+            "schema": {
+              "$ref": "#/definitions/smartGroup"
+            }
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "#/definitions/errorResponse"
+            }
+          }
+        },
+        "x-ms-examples": {
+          "Get": {
+            "$ref": "./examples/SmartGroups_GetById.json"
+          }
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/providers/Microsoft.AlertsManagement/smartGroups/{smartGroupId}/changeState": {
+      "post": {
```

Comment:
>post [](start = 7, length = 4)

Couldn't we use PUT itself to update the smartGroupState property ?

Reply:
same feedback. subscription level PUTs are allowed.. 
Please add PUT for create and Patch for updating the state
 


---
In reply to: [265821049](https://github.com/Azure/azure-rest-api-specs/pull/5384#discussion_r265821049) [](ancestors = 265821049)

Reply:
SmartGroups are also same as alerts. 

This is the link for RPSDK Interface for subscription wide resources. 
https://msazure.visualstudio.com/One/_git/AzureUX-ResourceProviderSDK?path=%2Fsrc%2FResourceProvider%2FRouting%2FRouting%2FResourceManagement%2FISubscriptionWideResourceTypeHandler.cs&version=GBdev

It contains only GET and POST. We can't change smart groups change state to PATCH. The API is already in public preview. If you want to discuss, we can schedule a meeting.

Reply:
If you want I can remove all other resources from this API version. This API version will be used only for alerts and alertsSummary.

Reply:
I have removed all other resources.